### PR TITLE
Policy to create a new hub

### PR DESCRIPTION
**Pull request by balajialg**
in response to issue #2670

### CODE REVIEW

File: docs/policy/create_policy.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Process to publish policy proposals
```
Comment (yuvipanda) [inline docs/policy/create_policy.rst:1]: This is using markdown syntax, so the file should have a .md extension

File: docs/policy/policy_create_hubs.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#Policy considerations while creating a new hub
+
+We have mastered creating new hubs based on our prior experience creating 10+ hubs catering to the diverse needs of the instructors across the campus. Our decisions with regards to creating a new hub were made with a lot of intuition around solving instructors' needs effectively within the time constraints.
+The objective of this policy document is to verbalize our heuristics for creating a new hub so that it can guide the decision making of all team members in the future. 
+
+- Course has a large user base with more than 300+ students (Eg: Data 8 and Data 100 hubs)
+- Course is computationally intensive and requires a large amount of CPU/Memory requirements (RAM greater than 2 GB or requires the use of calendar-based scheduling on a weekly basis) like a biology hub
+- Testbed for deploying new features which post maturity can be enabled across other major hubs (Eg: Stat 159 hub)
+- Course requirements tend to be dynamic and require making more than 5+ CI/CD deployments every month (Eg: Stat 159 hub)
```
Comment (yuvipanda) [inline docs/policy/policy_create_hubs.rst:9]: We should endeavor to make deployments as painless as possible, and I think this might create perverse incentives to say 'oh it is a high change hub, so it is less stable'. I think we should remove this one.

File: docs/policy/policy_create_hubs.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#Policy considerations while creating a new hub
+
+We have mastered creating new hubs based on our prior experience creating 10+ hubs catering to the diverse needs of the instructors across the campus. Our decisions with regards to creating a new hub were made with a lot of intuition around solving instructors' needs effectively within the time constraints.
```
Comment (yuvipanda) [inline docs/policy/policy_create_hubs.rst:3]:
```suggestion
We have a lot of experience in creating new hubs based on our prior experience creating 10+ hubs catering to the diverse needs of the instructors across the campus. Our decisions with regards to creating a new hub were made with a lot of intuition around solving instructors' needs effectively within the time constraints.
```

File: docs/policy/policy_create_hubs.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+The objective of this policy document is to verbalize our heuristics for creating a new hub so that it can guide the decision making of all team members in the future. 
+
+- Course has a large user base with more than 300+ students (Eg: Data 8 and Data 100 hubs)
```
Comment (yuvipanda) [inline docs/policy/policy_create_hubs.rst:6]: We should order these in order of importance, and basically say 'keep going down the list, and the sooner you hit a condition that is satisfied, the more likely you should get your own hub'.

Just to re-arrange this a bit, I think it should be:

1. Course needs image customization beyond what we currently have in the main datahub image. It doesn't necessarily need to be incompatible, but more than 'just this one package' - we don't want complexity. This is most likely the primary reason
2. Admin access
3. Testbed for deploying new features.
4. Computationaly intensive + calendar based scheduling.
5. Organizational reasons

File: docs/policy/policy_deployment_datahub.rst [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#Policy considerations while deploying to the main Datahub
+
+Our goal is to provide a reliable infrastructure that instructors can completely trust while using as part of their coursework. The objective of this policy document is to outline our criteria while deploying a change to the image in the main Datahub/other non-course-specific hubs (Like R Hub). 
+
+- Regular package changes during the start and end of the semester.
```
Comment (yuvipanda) [inline docs/policy/policy_deployment_datahub.rst]: Let's split this out into a different PR so we can discuss this separately?

**Review by yuvipanda — CHANGES_REQUESTED** [review]
Thanks a lot for working on this, Balaji! I've left a first series of comments. Do you have a local setup to build these docs to see how they look? It should be a fairly standard sphinx setup under docs/. I can help if needed.

File: docs/policy/create_policy.rst (hunk 1 above)
Comment (balajialg) [reply docs/policy/create_policy.rst:1]: Ah, got it!

File: docs/policy/policy_deployment_datahub.rst (hunk 5 above)
Comment (balajialg) [reply docs/policy/policy_deployment_datahub.rst]: Sounds good!

**Comment by balajialg** [comment]
@yuvipanda I addressed almost all of your comments. I will wait for a day to check if you or @felder have any inputs. If none,  by the same time tomorrow, will merge this to prod! Thanks :)

**Review by yuvipanda — CHANGES_REQUESTED** [review]
This looks great to me, @balajialg! I think a couple additional things to do:

1. Create a `policy/index.md` file listing the various policies, using a `toctree` directive
2. Link to `policy/index.md` from `index.rst` so it will show up on docs.datahub.berkeley.edu

Later on, we should move the storage-retention.rst policy file under here too. https://github.com/2i2c-org/infrastructure/tree/master/docs has some inspiration around the structure.

**Comment by balajialg** [comment]
@yuvipanda Sounds great! I added the toctree structure and linked it to index.rst file. Good learning experience :)

I will create another PR to discuss the policy for deploying to the main datahub. Will scope the storage retention policy relocation stuff as part of that PR

**Comment by balajialg** [comment]
@felder I realize that I didn't explicitly solicit any feedback from your end wrt this PR. Do you have any final thoughts? Are you aligned or do you expect me to revisit this policy proposal again as part of another PR? Will appreciate your input. Thanks!
